atlasdb-config/src/main/java/com/palantir/atlasdb/config/AtlasDbConfig.java

```diff
@@ -148,4 +149,15 @@ public long getSweepPauseMillis() {
     public int getSweepBatchSize() {
         return AtlasDbConstants.DEFAULT_SWEEP_BATCH_SIZE;
     }
+
+    @Value.Check
+    protected final void check() {
+        Preconditions.checkState(!leader().isPresent() || !lock().isPresent(),
+                "Leader and lock configuration blocks must not both be present.");
+        Preconditions.checkState(!leader().isPresent() || !timestamp().isPresent(),
+                "Leader and timestamp configuration blocks must not both be present.");
+
+        Preconditions.checkState(lock().isPresent() == timestamp().isPresent(),
+                "Can't only have one of lock and timestamp server blocks.");
```

Comment:
Nit: To me, these are easier to parse as positives, e.g. "Lock and timestamp server blocks must either both be present or both be absent"